### PR TITLE
Convert bundle importer to module

### PR DESCRIPTION
**Pull request by hatched**

### CODE REVIEW

File: jujugui/static/gui/src/app/init.js [hunk 1]
```diff
@@ -1619,7 +1620,7 @@ class GUIApp {
     this.db.destroy();
     this.endpointsController.destroy();
     // Detach event listeners.
-    const remove = document.removeEventListener;
+    const remove = document.removeEventListener.bind(document);
```
Comment (hatched) [inline jujugui/static/gui/src/app/init.js]: Without this, the `remove()` calls no longer have access to the proper `document` context so it never removes the events causing cascading test failures

File: jujugui/static/gui/src/app/init/bundle-importer.js [hunk 2]
```diff
@@ -0,0 +1,683 @@
+/* Copyright (C) 2017 Canonical Ltd. */
```
Comment (hatched) [inline jujugui/static/gui/src/app/init/bundle-importer.js]: No changes were made, just porting over and updating from var's to lets, etc.

File: jujugui/static/gui/src/app/init/test-bundle-importer.js [hunk 3]
```diff
@@ -1,42 +1,26 @@
-/*
```
Comment (hatched) [inline jujugui/static/gui/src/app/init/test-bundle-importer.js]: The tests were simply ported over as well.

File: jujugui/static/gui/src/app/utils/bakery-utils.js [hunk 4]
```diff
@@ -1,28 +1,13 @@
...
+/* Copyright (C) 2017 Canonical Ltd. */
 'use strict';
 
 /**
   Helper functions for creating and working with the bakery.
 */
 
 const Notification = require('../components/notification/notification');
+const React = require('react');
+const ReactDOM = require('react-dom');
```
Comment (hatched) [inline jujugui/static/gui/src/app/utils/bakery-utils.js]: Just a driveby looks like they were missed in @Lukewh 's branch which I had to land to continue on this...nbd!

File: jujugui/static/gui/src/app/utils/bundle-importer.js [hunk 5]
```diff
@@ -1,760 +0,0 @@
...
-SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
-General Public License for more details.
-
-You should have received a copy of the GNU Affero General Public License along
-with this program.  If not, see <http://www.gnu.org/licenses/>.
-*/
-
-'use strict';
-
-YUI.add('bundle-importer', function(Y) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/utils/bundle-importer.js]: bye bye more YUI

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8067/

File: jujugui/static/gui/src/test/factory.js [hunk 6]
```diff
@@ -18,7 +18,7 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 'use strict';
 
-YUI(GlobalConfig).add('juju-tests-factory', function(Y) {
+YUI.add('juju-tests-factory', function(Y) {
```
Comment (hatched) [inline jujugui/static/gui/src/test/factory.js]: This was wrong, has been wrong forever, not sure how it ever worked...

File: jujugui/static/gui/src/test/test_app.js [hunk 7]
```diff
@@ -1,2226 +0,0 @@
-/*
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_app.js]: We'll probably want to revisit this file to write the tests for the init, but it being here causes issues when trying to get a running suite.

File: jujugui/static/gui/src/test/test_autodeploy_extension.js [hunk 8]
```diff
@@ -1,185 +0,0 @@
-/*
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_autodeploy_extension.js]: These tests and the module functionality has been moved over to the init.

File: jujugui/static/gui/src/test/test_app.js (hunk 7 above)
Comment (huwshimi) [reply jujugui/static/gui/src/test/test_app.js]: I wonder if we could skip these tests instead of removing them, then we can remove the parts as we port them over to keep track of what has been done. Thoughts?

Comment (hatched) [reply jujugui/static/gui/src/test/test_app.js]: I figured that because we would be rewriting them and only making sure we cover the same functionality the code itself probably wouldn't be very helpful whereas having it in here could cause some issues with dependency loading that we don't need to deal with
